nncf/quantization/algorithms/smooth_quantize/backend.py

```diff
@@ -0,0 +1,205 @@
+        :param port_id: The input port ID to get weight input.
+        :return: The weight value.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_weight_tensor_port_id(node: NNCFNode) -> int:
+        """
+        Returns node's input port indices with weights tensors.
+
+        :param node: NNCFNode to find its weights input port indices.
+        :return: Weights input port indices.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def clip_statistics(statistics: TTensor) -> TTensor:
+        """
+        Clips statistics for further calculation.
+
+        :param statistics: Input statistics.
+        :return: Clipped statistics.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def calculate_scale_and_ratio(
+        activations: TTensor, weights: TTensor, alpha: float, quantile: Optional[float]
+    ) -> Tuple[TTensor, TTensor]:
+        """
+        Calculates base scale value and it's ratio.
+
+        :param activations: Activation statistics value.
+        :param weights: Weights statistics value.
+        :param alpha: Base value for exponentiation.
+        :param quantile: Base quantile value.
+        :return: Calculated base scale value & ratio.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def calculate_activation_scale(scale_value: TTensor, nodes: List[NNCFNode]) -> TTensor:
+        """
+        Calculates activation scales for Smooth node.
+
+        :param scale_value: Base scale value.
+        :param nodes: List of consumers for Smooth node.
+        :return: Calculated per-channel activation scale.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def calculate_weight_scale(scale_value: TTensor, nodes: List[NNCFNode]) -> TTensor:
+        """
+        Calculates scales for weight tensor.
+
+        :param scale_value: Base scale value.
+        :param nodes: List of consumers for Smooth node.
+        :return: Calculated per-channel scale for weights.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def weight_update_command(
+        node_with_weight: NNCFNode, weight_value: TTensor, weight_port_id: int
+    ) -> TransformationCommand:
+        """
+        Returns command to update weights.
+
+        :param node_with_weight: NNCFNode instance.
+        :param weight_value: New weight value.
+        :param weight_port_id: Weight port id.
+        :return: TransformationCommand instance.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def scale_insertion_command(
+        source_node: NNCFNode, scale_value: TTensor, port_id: int, nodes: List[NNCFNode]
+    ) -> TransformationCommand:
+        """
+        Returns command to insert Smooth Quant node.
+
+        :param source_node: NNCFNode instance.
+        :param scale_value: Smooth Quant value.
+        :param port_id: Output port for source node.
+        :param nodes: List of consumers for Smooth node.
+        :return: TransformationCommand instance.
+        """
```

Comment:
This backend has so many methods to implement, good case to inspect and work on NNCF as a Framework for algorithm implementation improvement

Reply:
Agree